src/utils/safe-messages.ts

```diff
@@ -47,8 +47,8 @@ export const supportsEIP1271 = ({ chainId, version, fallbackHandler }: SafeInfo)
 
   const isHandledByFallbackHandler = gte(version, EIP1271_FALLBACK_HANDLER_SUPPORTED_SAFE_VERSION)
   if (isHandledByFallbackHandler) {
-    const fallbackHandlerDeployment = getCompatibilityFallbackHandlerDeployment({ network: chainId, version })
-    return !!fallbackHandler.value && fallbackHandler.value === fallbackHandlerDeployment?.networkAddresses[chainId]
+    // We only check if any fallback Handler is set as we expect / assume that users who overwrite the fallback handler by a custom one know what they are doing
+    return fallbackHandler !== null && validateAddress(fallbackHandler.value) === undefined
```

Comment:
Should this not be `validateAddress(fallbackHandler.value) !== undefined` so that `supportsEIP1271` returns true if there is a fallbackhandler address set?

Reply:
`validateAddress` will return `undefined` or a validation error `string` 
So if the return value is `!== undefined` it would mean that some validation errors exist and the address is not valid.
While return  value `=== undefined` means that no validation error exists and the adress is valid.

Reply:
Ah I just realized `validateAddress` returns undefined if everything is ok with the address so it should be fine as it is.

Reply:
I will add a small util function "isValidAddress" so it reads better. This can be confusing.